jerry-core/ecma/base/ecma-globals.h

```diff
@@ -164,6 +164,15 @@ typedef int32_t ecma_integer_value_t;
   (ECMA_INTEGER_NUMBER_MIN <= (num) && (num) <= ECMA_INTEGER_NUMBER_MAX)
 
 /**
+ * Maximum integer number for multpily two ecma values
+ */
+#if CONFIG_ECMA_NUMBER_TYPE == CONFIG_ECMA_NUMBER_FLOAT32
+#define ECMA_INTEGER_MULTIPLY_MAX 0xb50
```

Comment:
The integer max is also different for float32 and 64 so the square root of max yield a different number for these modes.